src/rules/hostreq.ts

```diff
@@ -0,0 +1,52 @@
+import { Predicate, RuleMatchedAt, YAMLRule } from "../lint";
+import * as semver from "semver";
+import * as _ from "lodash";
+
+export class Semver implements Predicate<any> {
+  constructor(private readonly path: string,
+              private readonly required: boolean,
+  ) {}
+
+  public test(root: any): RuleMatchedAt {
+    const value = _.get(root, this.path);
+
+    if (!value && this.required) {
+      return { matched: true, paths: [] };
+    }
+
+    if (value && !semver.valid(value)) {
+      return { matched: true, paths: [this.path] };
+    }
+
+    return { matched: false };
+  }
+}
+
+export const hostreqSemverValid: YAMLRule = {
+  name: "prop-hostreq-version-semver-valid",
+  type: "error",
+  message: "replicated_version must be a valid semver specification",
+  test: new Semver("host_requirements.replicated_version", false),
+  examples: {
+    right: [{
+      yaml: `
+---
+host_requirements:
+  replicated_version: 2.9.0
```

Comment:
I tested this out, I think a single version is a valid "range":

```
package main

import (
	"fmt"
	"github.com/blang/semver"
)

func main() {
	_, err := semver.ParseRange("1.2.1")
	fmt.Println(err == nil); // true
}```

No sure if this would be an issue down the line, but at least as far as libyaml is concerned I think we tolerate a single version as a range here.